### ISSUE TITLE
draw PixelGrid rendering speed fix sugestion

### ISSUE THREAD

**Issue by LucasTheCure** [lbalazscs/Pixelitor#153]
I played with pixel grid, and this seems to work fine. 

```
private void drawPixelGrid(Graphics2D g2) {
        g2.setColor(WHITE); // this fix problem with rendering speed
        g2.setXORMode(BLACK);
        //... rest of the function body
}
```

**Comment by lbalazscs** [comment]
Great!
I assume you are referring to the Linux slowdown discussed in issue #140.
The Windows slowdown with selection (also described #140) didn't disappear from this change (pixel grid with selection can be enabled by taking out  !comp.showsSelection() from View.allowPixelGrid()), but if it becomes faster on Linux, then this is already an improvement. 
It also looks better this way, because there are no distracting colors in the pixel grid.

I didn't test on Linux yet, because while playing with the pixel grid I discovered some other issues in the current version (1. if there is a zoom, the selection is painted only for some tools (big problem)  2.  the pixel grid is painted above the tool handles (small problem, but annoying) ), I plan to have a deeper look at these issues this evening.

But anyway this is an improvement. If you want, send a pull request, or else I'll just include it in the next commit.

**Comment by LucasTheCure** [comment]
I test on Linux only. Confirm that still don't work properly when selection is on.

**Comment by lbalazscs** [comment]
I finally found out (after many years...) why the selection ruins the performance: it's setting a special Stroke on the Graphics, and it doesn't clear it. The pixel grid painting is using that slow Stroke because it's the same Graphics. Simply adding

```java
g2.setStroke(new BasicStroke());
```
under your code solves the problem with the selection. I am not commiting it yet, because I started some bigger changes to the painting system, but it should be ready soon.
